Loop/Managers/StatusExtensionDataManager.swift

```diff
@@ -134,6 +135,30 @@ final class StatusExtensionDataManager {
                 )
             }
             
+            updateGroup.enter()
+            manager.doseStore.insulinOnBoard(at: Date()) {(result) in
+                // This function completes asynchronously, so below
+                // is a completion that returns a value after eventual
+                // function completion.  Currently the time of update
+                // isn't used in the code, but could e.g. check how
+                // recent it is. 
+                switch result {
+                case .success(let iobValue):
+                    context.activeInsulin = ActiveInsulinContext(value: iobValue.value,
+                                                                 startDate: iobValue.startDate,
+                                                                 error: "")
+                    
+                case .failure(let error):
+                    // Use a large negative value to flag that we
```

Comment:
signaling with invalid values is a dangerous pattern, especially in systems like this. If we're concerned about the possible downstream uses of this value (and we should be) then I think the only safe thing to do in an error situation is to not return an ActiveInsulinContext.

So, instead of making error an optional String (as I suggested elsewhere), you can leave context.activeInsulin nil in case of error. Since we know that this value is expected in this version of the code, the fact that it's nil indicates that there's an error. That will simplify your downstream code and avoid fake values.

Reply:
👍

Reply:
OK, agreed with all of the above.  I will tackle these changes soon.  Thanks for the feedback!